File: test/12_test.dart
import 'dart:io';
import 'package:test/test.dart';
import 'package:advent_of_code_2020/12.dart';

void main() {
  String contents = new File('./data/12.txt').readAsStringSync();

  String example = 'F10\nN3\nF7\nR90\nF11';
  
  group('A', () {
    test('Example 1', () {
      expect(A12(example), equals(25));
    });
    // test('Solution', () {
    //   expect(A12(contents), equals(582));
    // });
  });

  // group('B', () {
  //   test('Example 1', () {
  //     expect(B12(example), equals(286));
  //   });
  //   test('Solution', () {
  //     expect(B12(contents), equals(52069));
  //   });
  // });
}

File: test/12.test.js
const fs = require('fs');

const { A12, B12 } = require('../lib/12');
const input = fs.readFileSync('./data/12.txt', 'utf-8');

const example = 'F10\nN3\nF7\nR90\nF11';

describe('12', () => {
	describe('A', () => {
		it('Example 1', () => {
			expect(A12(example)).toBe(25);
		});

		it('Solution', () => {
			expect(A12(input)).toBe(582);
		});
	});

	describe('B', () => {
		it('Example 1', () => {
			expect(B12(example)).toBe(286);
		});

		it('Solution', () => {
			expect(B12(input)).toBe(52069);
		});
	});
});
